### PR TITLE
Bug 1353196 - Fix malformed categorical histograms

### PR DESCRIPTION
**Pull request by fbertsch**

### CODE REVIEW

File: moztelemetry/histogram.py [hunk 1]
```diff
@@ -133,7 +133,10 @@ def __init__(self, name, instance, revision=None, histograms_url=None,
             pd_index = labels + [CATEGORICAL_HISTOGRAM_SPILL_BUCKET_NAME]
             if isinstance(instance, dict):
                 # pandas clobbers values if instance is a dict with string labels
-                entries = {int(k): v for k, v in instance["values"].items()}
+                try:
+                    entries = {int(k): v for k, v in instance["values"].items()}
```
Comment (vitillo) [inline moztelemetry/histogram.py]: Can we do the same also for non-categorical histograms?

Comment (fbertsch) [inline moztelemetry/histogram.py]: Done and done.

**Comment by fbertsch** [comment]
@vitillo RFAL

File: moztelemetry/histogram.py (hunk 1 above)
Comment (vitillo) [inline moztelemetry/histogram.py]: Genera comment not related to this patch: this code is pretty ugly, would be nice to refactor the initialization logic for the different histogram types in separate methods.

Comment (fbertsch) [inline moztelemetry/histogram.py]: Agree wholeheartedly. The Histogram class as a whole could use some love.

**Pull request merged**